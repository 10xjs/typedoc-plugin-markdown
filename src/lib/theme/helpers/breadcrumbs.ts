import { Reflection } from 'typedoc';
import { PageEvent } from 'typedoc/dist/lib/output/events';
import { MarkdownPlugin } from '../../plugin';
import { DocusaurusTheme } from '../theme.docusaurus';
import { GitbookTheme } from '../theme.gitbook';
import { relativeUrl } from './relative-url';

export function breadcrumbs(this: PageEvent) {
  if (!isVisible()) {
    return '';
  }
  const md = [`[Globals](${relativeUrl(this.project.url)}) /`];
  return breadcrumb(this.model, md);
}

function breadcrumb(model: Reflection, md: string[]) {
  if (model && model.parent) {
    breadcrumb(model.parent, md);
    if (model.url) {
      md.push(`[${model.name}](${relativeUrl(model.url)}) /`);
    }
  }
  return md.join(' ');
}

function isVisible() {
  if (MarkdownPlugin.theme instanceof DocusaurusTheme || MarkdownPlugin.theme instanceof GitbookTheme) {
    return false;
  }
  return true;
}
