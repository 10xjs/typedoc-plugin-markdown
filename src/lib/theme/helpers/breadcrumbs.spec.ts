import { breadcrumbs } from './breadcrumbs';

describe(`breadcrumbs helper`, () => {
  test(`should compie`, () => {
    const page = {
      project: { name: 'typedoc-plugin-markdown', url: 'globals.md' },
      model: { name: 'Animal', parent: { name: 'Classes', url: 'modules/_classes_.md', parent: {} }, url: 'classes/_classes_.animal.md' },
    };
    expect(breadcrumbs.call(page)).toMatchSnapshot();
  });
});
