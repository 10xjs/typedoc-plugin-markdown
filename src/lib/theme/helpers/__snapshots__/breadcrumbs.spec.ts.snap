// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breadcrumbs helper should compie 1`] = `"[Globals](globals.md) / [Classes](modules/_classes_.md) / [Animal](classes/_classes_.animal.md) /"`;
